kitsune/users/api.py

```diff
@@ -234,25 +204,6 @@ def validate(self, data):
 
         return data
 
-    def create(self, validated_data):
-        user_data = validated_data.pop("user")
-        u = RegistrationProfile.objects.create_inactive_user(
-            user_data["username"], user_data["password"], user_data["email"]
-        )
-        p = u.profile
-        for key, val in validated_data.items():
-            setattr(p, key, val)
-        p.save()
-        return p
-
-    def update(self, instance, validated_data):
-        if "user" in validated_data:
-            user_data = validated_data.pop("user")
-            for key, val in user_data.items():
-                setattr(instance.user, key, val)
-            instance.user.save()
-        return super(ProfileSerializer, self).update(instance, validated_data)
-
     def validate_username(self, username):
```

Comment:
As this serializer should not be used for creating profile anymore, you should remove the validation as well.